src/bun.js/test/jest.zig

```diff
@@ -3821,7 +3829,101 @@ pub const Expect = struct {
         return .zero;
     }
 
-    pub const toHaveBeenCalledTimes = notImplementedJSCFn;
+    pub fn toHaveBeenCalled(this: *Expect, globalObject: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSC.JSValue {
+        const thisValue = callframe.this();
+        defer this.postMatch(globalObject);
+
+        const value: JSValue = JSC.Jest.Expect.capturedValueGetCached(thisValue) orelse {
+            globalObject.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
+            return .zero;
+        };
+
+        const calls = JSMockFunction__getCalls(value);
+        active_test_expectation_counter.actual += 1;
+
+        if (calls == .zero or !calls.jsType().isArray()) {
+            globalObject.throw("Expected value must be a mock function: {}", .{value});
```

Comment:
The `if` check and error message don't seem to match up?

Reply:
that is how it detects if it's a mocked callback